### DISCUSSION TITLE
Connect to GitHub/GitLab via ssh

### DISCUSSION THREAD

**Discussion by Msadat97** [coder/code-server#5867] [Q&A] [answered]
Hi all,

I cannot connect to GitHub/GitLab via `ssh` in the code-server terminal and could not find a solution so far. When I try `ssh -Tv -p 22 git@github.com`, it gives me `ssh: connect to host github.com port 22: Connection refused`. 

My setup:

1. Request a compute node via SLURM and attach a code-server to the node.
2. Run git command inside the code-server

Changing the port does not sovle my issue.

Thank you in advance for your help.

**Comment by jsjoeio** [comment]
Hmm 🤔 any ideas @code-asher ?

**Reply by code-asher** [reply]
Does SLURM block external network access?

**Comment by lg-nineoneninethreenine** [comment]
Let me suggest this: make and add your SSH key for authentication.

I was trying out Gitea last week, which is similar to this. In guides, they said that to clone repos via SSH, you should add your SSH key to do so. I think it's the same here, and that @Msadat97 might have skipped that step.

Best way?

Use ssh-keygen to make a key that is compatible with GitHub. I think it uses ED25519 for SSH authentications. Make it with a title and passphrase is needed, add the public and private key to GitHub and test it out. There should be docs too, but honestly, who would look at that right now during 11 PM here?

Try a Google search if you need more info, I told whatever I knew. Sorry if my answer wasn't good enough, it is just exhausting right now in my life.

**Reply by lg-nineoneninethreenine** [reply]
did it work

**Comment by Msadat97** [comment, marked as answer]
Hi all.

Many Thanks for the replies.

I realized that the git SSH option is blocked by SLRUM in our cluster. So there is nothing we can do to solve that.
